splinepy/nurbs.py

```diff
@@ -155,50 +101,28 @@ def _update_p(self,):
             "now identical."
         )
 
-    def copy(self,):
+    def extract_bezier_patches(self):
         """
-        Returns freshly initialized Nurbs of self.
+        Extract all knot spans as Bezier patches to perform further operations
+        such as compositions and multiplications
 
         Parameters
-        -----------
+        ----------
         None
 
-        Returns
-        --------
-        new_nurbs: `NURBS`
-        """
-        new_nurbs = NURBS(**self.todict())
-
-        return new_nurbs
-
-    def todict(self, tolist=False):
+        Returns 
+        -------
+        extracted Beziers : list
         """
-        Returns copy of degrees, knot_vectors, control_points, weights in dict.
-
-        Parameters
-        -----------
-        tolist : bool
-          Default is False. Convert numpy properties into lists
-
-        Returns
-        --------
-        dict_spline: dict
-          Keys are {degrees, knot_vectors, control_points, weights}.
-        """
-        if tolist:
-            return dict(
-                degrees=self.degrees.tolist(),
-                knot_vectors=self.knot_vectors.tolist(),
-                control_points=self.control_points.tolist(),
-                weights=self.weights.tolist()
-            )
-        else:
-            return dict(
-                degrees=copy.deepcopy(self.degrees),
-                knot_vectors=copy.deepcopy(self.knot_vectors),
-                control_points=copy.deepcopy(self.control_points),
-                weights=copy.deepcopy(self.weights)
-            )
-
-    # member function alias
-    ws = weights
+        # Extract bezier patches and create PyRationalBezier objects
+        list_of_c_object_beziers = self._c_spline.extract_bezier_patches()
+
+        # Transform into Rational Bezier Splinepy objects
+        extracted_bezier_patches = []
+        for c_object_spline in list_of_c_object_beziers:
+            i_rational_bezier = RationalBezier()
+            i_rational_bezier._c_spline = c_object_spline
+            i_rational_bezier._update_p()
+            extracted_bezier_patches.append(i_rational_bezier)
+
+        return extracted_bezier_patches
```

Comment:
Just as a reminder for discussion: would it be nice to have an intermediate abstractclass?